src/script/lua_api/l_metadata.cpp

```diff
@@ -177,7 +177,7 @@ int MetaDataRef::l_get_float(lua_State *L)
 
 	std::string str_;
 	const std::string &str = meta->getString(name, &str_);
-	lua_pushnumber(L, stof(str));
+	lua_pushnumber(L, atof(str.c_str()));
```

Comment:
this would be the lua built-in way
```suggestion
	lua_pushstring(L, str.c_str());
	auto r = lua_tonumber(L, -1);
	// returns 0 on error but it's also a valid value
	if (r == 0 && lua_type(L, -1) == LUA_TSTRING) {
		lua_pop(L, 1);
		lua_pushnumber(L, 0);
	}
```

Reply:
This would introduce a subtle incompatibility: `atof` parses e.g. "5 with suffix" as 5, while `lua_tonumber` returns 0 when given the same string.

Reply:
Do we want to support that?

Reply:
Fair enough.